dotcom-rendering/src/lib/content.d.ts

```diff
@@ -286,6 +286,11 @@ interface MultiImageBlockElement {
 	role?: RoleType;
 }
 
+interface NewsletterSignupBlockElement {
+	_type: 'model.dotcomrendering.pageElements.NewsletterSignupBlockElement';
+	newsletter: NewsletterData;
```

Comment:
Can we inline these properties here? The other block element types mostly follow this same convention and it would be good to keep this pattern consistent

Reply:
Sounds good - I'm keen to keep `NewsletterData` as a type so it aligns with the object we are going to generate in Frontend, so I used `interface NewsletterSignupBlockElement extends NewsletterData` - hope that's ok.

Reply:
I actually think in this case it might be better to not try to align. The reason I say this is because we have no solution at the moment for sharing types between Frontend and DCR, this is a known issue and we've discussed various options (GraphQL, Thrift, etc.). This is very much a long term project and until we have a code based solution it is not possible to couple types between the two platforms.

Of course, that doesn't stop us from creating one type in Frontend and the same one in DCR but I would say doing this - without technical constraints - would create more problems. In that scenario it is very easy to edit one or the other type and for the other one to get out of alignment. Once this happens we will have code that is 'meant to be' the same but isn't 😬 

Instead of this, in DCR, we consciously made the choice to have completely different models. This is why all the other element blocks have types that are subsets or constructions formed of other data from Frontend. Like this you are free to make changes at either end without creating confusion.

Note. This was the route we took for DCR. On the apps rendering project they went the _other_ way, they pass in the data from CAPI directly to the rendering tier, so it's not that one method is right or wrong, just that this is the convention DCR follows.

Sorry for the long rambling comment!